backend/canisters/community/impl/src/updates/c2c_join_channel.rs

```diff
@@ -51,7 +51,16 @@ async fn c2c_join_channel(args: Args) -> Response {
     }
 }
 
-pub(crate) async fn join_channel_impl(channel_id: ChannelId, user_principal: Principal) -> Response {
+pub(crate) fn join_channel_auto(channel_id: ChannelId, user_principal: Principal) {
+    match read_state(|state| is_permitted_to_join(channel_id, user_principal, state)) {
+        Ok(None) => {}
+        _ => return,
+    };
+
+    mutate_state(|state| commit(channel_id, user_principal, state));
+}
+
+async fn join_channel_internal(channel_id: ChannelId, user_principal: Principal) -> Response {
```

Comment:
Maybe this should be `check_gate_then_join_channel`?

Reply:
Its not very obvious what the difference is otherwise

Reply:
ok